core/src/main/java/org/infinispan/iteration/impl/TransactionAwareCloseableIterator.java

```diff
@@ -104,7 +104,7 @@ public TransactionAwareCloseableIterator(CloseableIterator<CacheEntry<K, V>> rea
       if (returnedEntry == null) {
          // We do a last check to make sure no additional values were added to our context while iterating
          for (CacheEntry lookedUpEntry : ctx.getLookedUpEntries().values()) {
-            if (seenContextKeys.add(lookedUpEntry.getKey()) && !lookedUpEntry.isRemoved()) {
+            if (seenContextKeys.add(lookedUpEntry.getKey()) && !lookedUpEntry.isRemoved() && !lookedUpEntry.isNull()) {
```

Comment:
It can't be null there.  Those are entries returned from a remote node or local data container.